cdf-core/cdf/js-modules/components/filter/basebone/basebone.js

```diff
@@ -0,0 +1,39 @@
+/* jshint devel:true */
+
+define([
+  'amd!cdf/lib/underscore',
+  'amd!cdf/lib/backbone',
+  'cdf/components/BaseComponent'],
+  function( _, Backbone, BaseComponent ) {
+
+    var rest = _.rest;
+
+    function extendClass(TargetClass) {
+      return BaseComponent.extend.apply(TargetClass, rest(arguments));
+    }
+
+    function addSelfExtend(TargetClass) {
+      return extendClass(TargetClass, {}, {
+        extend: BaseComponent.extend
+      });
+    }
+
+    function addEvents(TargetClass) {
+      return extendClass(TargetClass, Backbone.Events);
+    }
+
+    function basebonify(TargetClass) {
+      return extendClass(addEvents(addSelfExtend(TargetClass)), arguments[1], arguments[2]);
+    }
+
+    // BaseBone: returns Base.js modification that includes Backbone.Events.
+    //   Also has several static helpers to augment constructors with .extend
+    //   and events functionality.
+    var BaseBone = basebonify( BaseComponent );
+
```

Comment:
Oops, my bad, forgot to update this. I'll rename basebonify -> convertClass. 
@pamval: do you object to the name "BaseBone" as well?